courseuserbadges.php

```diff
@@ -85,24 +90,21 @@
         require_capability('local/obf:viewhistory', $context);
         $client = obf_client::get_instance();
 
-        $search_params = array(
+        $search = optional_param('search', null, PARAM_TEXT);
+        $searchparams = array(
             'api_consumer_id' => OBF_API_CONSUMER_ID,
             'log_entry' => '"course_id":"' . $courseid . '"',
-            'count_only' => 1
+            'query' => $search
         );
-        $res = $client->get_assertions(null, null, $search_params);
-
-        $historysize = $res[0]['result_count'];
 
-        $search_params['count_only'] = 0;
-        $search_params['limit'] = 10;
-        $search_params['offset'] = $curr_page * 10;
-        $search_params['order_by'] = 'asc';
+        $searchparams['count_only'] = 0;
+        $searchparams['order_by'] = 'asc';
 
-        $history = obf_assertion::get_assertions($client, null, null, -1, false, $search_params);
+        $history = obf_assertion::get_assertions($client, null, null, -1, false, $searchparams);
+        $historysize = count($history);
```

Comment:
If I remember correctly, these changes were made in response to an issue observed during testing. It was discovered that user data of individuals not registered on the platform was being returned via the API request. As a result, due to GDPR regulations, it was requested that this data should not appear in any way on the lists and exports related to this feature. One of the workarounds I found involved retrieving all the API results at once and then sorting/paginating them later at the plugin level.

Reply:
Ok, I see. Getting all the rows at once doesn't work because some clients can have so many that it takes too long. You could do that with multiple api calls before presenting the data but it wastes bandwidth and probably causes timeouts.

I suggest that the pagination is done at the api level and then the plugin checks the results and any personal data that shouldn't be shown is masked.

Regarding the GDPR issue, there is actually a setting in local/obf/config.php that controls if OBF_API_CONSUMER_ID should be used in filters. ([see here](https://github.com/DigiDago/moodle-local_obf/blob/a92c991df1e6d5eeae176ac2a5ec3e4e6014bdd9/classes/client.php#L630), for example). It should probably be always enabled. When it's used, the results should only have badges where the issuing platform is "Moodle".

Reply:
Regarding this part, I have rolled back to the previous version of the code and now using API pagination. On the plugin side, we will now hide users who are no longer present on the platform, and on the display side we will use the string 'User removed' to mask those data.

However, for the side effect, we cannot perform a search using the "/v1/event" request because it does not support the "query" parameter. Do you think it's possible to add this parameter for this request? The idea is to enable searching based on the "Badge name" and "Badge description" data using "query" param juste like "/v1/badge/".

On the plugin side, I have temporarily disabled the search field for the relevant feature so that we can still release a v1 following Emilie's deadline, which is set for Friday.

Reply:
Thanks, I will make a note of the query parameter for further api development.